src/others/file_explorer/lv_file_explorer.c

```diff
@@ -690,12 +729,13 @@ static void sort_by_file_kind(lv_obj_t * tb, int16_t lo, int16_t hi)
     int16_t lt = lo;
     int16_t i = lo + 1;
     int16_t gt = hi;
-    const char * v = lv_table_get_cell_value(tb, lo, 1);
+    // const char * v = lv_table_get_cell_value(tb, lo, 1);
+    lv_file_explorer_file_table_entry_data_t * v = lv_table_get_cell_user_data(tb, lo, 0);
     while(i <= gt) {
-        int strcmp_result = lv_strcmp(lv_table_get_cell_value(tb, i, 1), v);
-        if(strcmp_result < 0)
+        lv_file_explorer_file_table_entry_data_t * x = lv_table_get_cell_user_data(tb, i, 0);
+        if(x->file_kind < v->file_kind)
             exch_table_item(tb, lt++, i++);
-        else if(strcmp_result > 0)
+        else if(x->file_kind > v->file_kind)
```

Comment:
My code sorts by the file type which is the same behavior as the original code. I could add sorting by name in a separate PR

Reply:
Oh I see, before it was sorting by file kind and name because the icon is part of the string.

It used to sort by name and now it doesn't, though. Is that right? The `lv_strcmp` compared the full entry name, not just the type.

Reply:
The files were sorted on the second column of the table(index 1) in the old code:
```c
int strcmp_result = lv_strcmp(lv_table_get_cell_value(tb, i, 1), v);
```
File names were stored in index 0 and file kinds in index 1 so the files were indeed sorted by file kind.

Reply:
> File names were stored in index 0 and file kinds in index 1 so the files were indeed sorted by file kind.

Ah I see, got it.